### PR TITLE
add video models

### PR DESCRIPTION
**Pull request by MatthieuRipollPassCulture**
# DA PR

## Describe your changes

Please include a summary of the changes: This PR adds 3 models related to the video modules displayed in app
- int_video_events, a raw table to only select video-related events
- int_firebase__native_daily_user_video_module, an intermediate table at video granularity
- mrt_native__daily_video_module_conversion, a mart table at video block granularity for dashboarding purpose

Tag a reviewer if necessacy  @carolinemestre-passculture @Jules-Arbelot 
## Jira ticket number and/or notion link

JIRA-ticket_number

### Type of change
- [ ] Fix (non-breaking change which corrects expected behavior)
- [ ] New fields (non-breaking change)
- [x] New table (non-breaking change)
- [ ] Concept change (potentially breaking change which modifies fields according to new or evolving business concepts) 
- [ ] Table deletion (potentially breaking change which adds functionality/ table)
      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] Fields have been snake_cased
- [x] I have checked my modifications don't break downstream models
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I have updated the dag in cases of dependencies
- [ ] My code passes CI/CD tests
- [x] I will post on slack review channel and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### CODE REVIEW

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{
+    config(
+        materialized = "incremental",
+        incremental_strategy = "insert_overwrite",
+        partition_by = {"field": "event_date", "data_type": "date", "granularity" : "day"},
+        on_schema_change = "sync_all_columns",
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql:6]: Peut-être mieux de laisser la synchro manuelle lors de l'ajout de colonnes pour ne pas tout rejouer sans le savoir

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    , module_id
+    , video_id
+    , entry_id
+    , ne.offer_id
+    , COALESCE(CAST(video_seen_duration_seconds AS FLOAT64),0) video_seen_duration_seconds -- A nettoyer en raw 
+    , COALESCE(CAST(video_duration_seconds AS FLOAT64),0) video_duration_seconds -- A nettoyer en raw
+FROM {{ref('int_firebase__native_event')}} ne
+INNER JOIN {{ ref('int_contentful__entry' )}}  ce ON ne.module_id = ce.id 
+                                            AND ce.content_type IN ('video','videoCarousel', 'videoCarouselItem')
+WHERE event_name IN ('ConsultVideo','HasSeenAllVideo', 'HasDismissedModal', 'VideoPaused', 'ModuleDisplayedOnHomePage', 'ConsultOffer')
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql:26]: ConsultOffer where origin = "VideoModal" pour compter uniquement les consultations qui viennent des vidéos ?

File: orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{{
+    config(
+        materialized = "incremental",
+        incremental_strategy = "insert_overwrite",
+        partition_by = {"field": "event_date", "data_type": "date", "granularity" : "day"},
+        on_schema_change = "sync_all_columns",
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql:6]: sûr de vouloir sync_all_columns ?

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+SELECT 
+    module_id
+    , event_date
+    , app_version
+    , entry_id
+    , user_id
+    , user_role
+    , unique_session_id    
+    , COUNT(DISTINCT video_id) AS total_videos_seen
+    , SUM(offers_consulted) AS offers_consulted
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql:53]: total_offers_consulted

File: orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    , user_role
+    , entry_id
+    , app_version
+    , COUNT(DISTINCT unique_session_id) AS total_session_display
+    , COUNT(DISTINCT CASE WHEN offers_consulted > 0 THEN unique_session_id ELSE NULL END) AS total_session_consult_offer
+    , SUM(offers_consulted) AS total_offers_consulted
+    , SUM(video_seen_duration_seconds) AS total_video_seen_duration_seconds
+    , COUNT(DISTINCT CASE WHEN pct_video_seen >= 0.25 THEN unique_session_id ELSE NULL END) AS total_seen_25_pct_video
+    , COUNT(DISTINCT CASE WHEN pct_video_seen >= 0.5 THEN unique_session_id ELSE NULL END) AS total_seen_50_pct_video
+    , COUNT(DISTINCT CASE WHEN pct_video_seen >= 0.75 THEN unique_session_id ELSE NULL END) AS total_seen_75_pct_video
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql:22]: Ajouter le total_seen_whole_video ?

File: orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+FROM {{ref('int_firebase__native_daily_user_video_module')}}
+    {% if is_incremental() %}
+    AND event_date BETWEEN date_sub(DATE("{{ ds() }}"), INTERVAL 1 DAY) and DATE("{{ ds() }}")
+    {% endif %}
+GROUP BY 1,2,3,4,5
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql:27]: Nommer les colonnes dans le group by

File: orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql (hunk 5 above)
Comment (Jules-Arbelot) [reply orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql:22]: Ajouter le total_video_duration

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    , SUM(video_seen_duration_seconds) AS video_seen_duration_seconds
+    , MAX(pct_video_seen) AS pct_video_seen
+FROM displays
+LEFT JOIN video_perf_per_user_and_video USING(module_id, entry_id, unique_session_id)
+GROUP BY 1,2,3,4,5,6,7
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql:58]: Nommer les colonnes dans le group by

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    , event_date
+    , app_version
+    , entry_id
+    , user_id
+    , user_role
+    , unique_session_id    
+    , COUNT(DISTINCT video_id) AS total_videos_seen
+    , SUM(offers_consulted) AS offers_consulted
+    , SUM(video_seen_duration_seconds) AS video_seen_duration_seconds
+    , MAX(pct_video_seen) AS pct_video_seen
```
Comment (Jules-Arbelot) [inline orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql:55]: Ajouter video_total_duration_seconds

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql (hunk 1 above)
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql:6]: Vu avec Jules : les deux stratégies sont ok à vous de décider laquelle vous préférez adopter

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql (hunk 8 above)
Comment (carolinemestre-passculture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql:55]: total_video_duration_seconds

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql (hunk 2 above)
Comment (MatthieuRipollPassCulture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase_native_video_event.sql:26]: Tu as plusieurs valeurs potentielles du paramètre origin ("VideoModal", "VideoCarrouselBlock" etc) mais le INNER JOIN sur contentful_entry te permet de t'assurer que tu ne prends que des events en lien avec les blocs vidéos

File: orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql (hunk 5 above)
Comment (MatthieuRipollPassCulture) [reply orchestration/dags/data_gcp_dbt/models/mart/native/mrt_native__daily_video_module_conversion.sql:22]: Je ne sais pas à quel point le total_video_duration ici est pertinent puisqu'on est à l'échelle du bloc (plusieurs vidéos)

File: orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+FROM displays
+LEFT JOIN video_perf_per_user_and_video USING(module_id, entry_id, unique_session_id)
```
[the remark this answers is not in the document]
Comment (MatthieuRipollPassCulture) [reply orchestration/dags/data_gcp_dbt/models/intermediate/firebase/native/int_firebase__native_daily_user_video_module.sql:57]: Nope pas censé

**Review by Jules-Arbelot — APPROVED** [review]

**Pull request merged**